tests/integration_test.py

```diff
@@ -0,0 +1,142 @@
+        self.archdirs = {}
+
+        for arch in ["x86", "amd64"]:
+            self.archdirs[arch] = os.path.join(self.test_dir, arch)
+            os.mkdir(self.archdirs[arch])
+
+        self.my_dir = os.path.dirname(__file__)
+        self.mcsema_lift = os.path.realpath(
+            os.path.join(self.my_dir, "..", "build", "mcsema-lift"))
+        self.mcsema_gencode = os.path.realpath(
+            os.path.join(self.my_dir, "..", "generated"))
+
+        try:
+            clang_file = subprocess.check_output(["which", "clang-3.8"])
+            self.clang = clang_file.strip()
+        except OSError as oe:
+            sys.stderr.write("Could not find clang-3.8: {}\n".format(str(oe)))
+            self.assertTrue(False)
+        except subprocess.CalledProcessError as ce:
+            sys.stderr.write("Could not find clang-3.8: {}\n".format(str(ce)))
+            self.assertTrue(False)
+
+
+    def tearDown(self):
+        # Remove the directory after the test
+        if not DEBUG:
+            shutil.rmtree(self.test_dir)
+
+    def _sanityCheckFile(self, fname):
+        self.assertTrue(os.path.exists(fname))
+        self.assertTrue(os.path.getsize(fname) > 0)
+
+    def _runWithTimeout(self, procargs, timeout=1200):
+
+        with open(os.devnull, "w") as devnull:
+            if DEBUG:
+                sys.stderr.write("executing: {}\n".format(" ".join(procargs)))
+            po = subprocess.Popen(procargs, stderr=devnull, stdout=devnull)
+            secs_used = 0
+
+            while po.poll() is None and secs_used < timeout:
+                time.sleep(1)
+                sys.stderr.write("~")
+                secs_used += 1
+
+        # took less than timeout
+        self.assertTrue(secs_used < timeout)
+
+        # successfully exited
+        self.assertTrue(po.returncode == 0)
+        sys.stderr.write("\n")
+
+    def _runAMD64Test(self, testname, entrypoint="main", buildargs=None):
+        self._runArchTest("amd64", testname, entrypoint, buildargs)
+
+    def _runX86Test(self, testname, entrypoint="main", buildargs=None):
+        self._runArchTest("x86", testname, entrypoint, buildargs)
+
+    def _compileBitcode(self, arch, infile, outfile, extra_args=None):
+
+        generated_asm = {
+                "amd64": "ELF_64_linux.S",
+                "x86": "ELF_32_linux.S",}
+
+        asm_file = os.path.join(self.mcsema_gencode, 
+                generated_asm[arch])
+
+        flags = {
+                "amd64": "-m64",
+                "x86": "-m32", }
+
+        self.assertTrue(os.path.exists(self.clang))
+        args = [self.clang,
+                flags[arch],
+                "-o", outfile,
+                asm_file,
+                infile,
+                ]
+
+        if extra_args:
+            args.extend(extra_args)
+
+        self._runWithTimeout(args)
+        self._sanityCheckFile(outfile)
+
+    def _runArchTest(self, arch, testname, entrypoint, buildargs=None):
```

Comment:
Going to do something similar: the code already lives in the LinuxTest class, but the Linux integration test can be run from another os (Windows, for instance). In that case, we shouldn't try to rebuild binaries since they wont run. Going to add a check to see if we are running Linux tests on Linux, and only then rebuild the binaries.